src/types.rs

```diff
@@ -0,0 +1,50 @@
+//! Concrete node and edge types used in the registry.
+
+/// The type of a node.
+pub enum NodeType {
+    /// A user, eg. contributor, project member etc.
+    User,
+    /// A project with users as members and contributors.
+    Project,
+}
+
+/// Node data.
+type NodeData = NodeType;
+
+/// The type of an edge.
+pub enum EdgeType {
+    /// Contribution from a user to a project. This edge type is
+    /// used bi-directionally, between project and user.
+    Contribution,
+    /// Membership of a user in a project, project <-> user.
+    ProjectMembership,
+    /// One-way dependency between two projects, project -> project.
+    ProjectDependency,
+}
+
+/// Edge data.
+pub struct EdgeData {
+    /// The type of edge.
+    edge_type: EdgeType,
+    /// A strength factor for this edge. Eg. the number of contributions
+    /// on a contribution edge.
+    strength: f64,
+}
+
+/// The rank or "osrank" of a node, normalized to `1.0`.
+pub type NodeRank = f64;
+
+/// Global parameters used by the graph algorithm.
+pub struct HyperParameters {
+    /// Threshold below which nodes are pruned in the first
+    /// phase of the algorithm.
+    pruning_threshold: f64,
```

Comment:
Can we also add a comment that `pruning_threshold` is effectively `tau`? It's a minor thing but we are now using `tau` both for our `osrank` code and for the specification (the "end to end document" so to speak) so it would be nice to establish a correlation.